nodebuilder/core/config.go

```diff
@@ -41,3 +41,10 @@ func (cfg *Config) Validate() error {
 	}
 	return nil
 }
+
+// EndpointConfigured returns whether a core endpoint has been set
+// on the config.
+func (cfg *Config) EndpointConfigured() bool {
+	defaultCfg := DefaultConfig()
+	return !(cfg.IP == defaultCfg.IP && (cfg.GRPCPort == defaultCfg.GRPCPort || cfg.RPCPort == defaultCfg.RPCPort))
+}
```

Comment:
The defaults are zeros. Do you remember why we have zeroes isntead of empty strings? We should consider emptying them to avoid this monstrosity.

Reply:
I think the defaults were 0s precisely because we did not provide a stubbed accessor and still attempted to dial the empty addr that was passed inside core_accessor.

Changed to empty strings now. https://github.com/celestiaorg/celestia-node/pull/2577/commits/1b2435685721246bc28d3380da8453d9b28fb745